Modules/_ctypes/_ctypes.c

```diff
@@ -2830,6 +2856,16 @@ PyCData_GetContainer(CDataObject *self)
     return self;
 }
 
+static CDataObject *
+PyCData_GetContainer(CDataObject *self)
```

Comment:
This isn't correct, it needs to acquire critical section after looping 
```c
    while (self->b_base) {
        self = self->b_base;
    }
```

I suggest to leave this fix for now and only focus on getting rid of LOCK_PTR and memcpy in this PR.